h/views/main.py

```diff
@@ -57,6 +57,15 @@ def robots(context, request):
 
 @view_config(route_name='stream')
 def stream(context, request):
+    if request.feature('search_page'):
+        q = request.params.get('q', '').split(':')
+        if len(q) == 2 and q[0] == 'tag':
```

Comment:
I just noticed that this isn't bullet-proof. I clicked on a tag that is "hypothesis:education", which then obviously didn't redirect because the length should be 3. Maybe we can change this to be `len(q) >= 2`?